### PR TITLE
PR: Add a syntax highlighter for Python log files

### PR DESCRIPTION
**Pull request by ccordoba12**
The idea is to see the log files generated by the Python language and client servers with syntax highlighting in Spyder itself.

Unfortunately, I didn't manage to get things working (after finding the code for this highlighter in an old Pygments PR).

@dalthviz, please help me to finish this PR. I think the easiest thing to do that is to use the highlighter through the terminal for some log files (if you start Spyder like `python bootstrap.py --debug`, you'll find the PyLS log files in `~/.spyder-py3-dev/lsp_logs`)

### CODE REVIEW

File: spyder/utils/syntaxhighlighters.py [hunk 1]
```diff
@@ -1230,6 +1220,37 @@ def highlightBlock(self, text):
             self.highlight_spaces(text)
 
 
+class PythonLoggingLexer(RegexLexer):
+    """
+    A lexer for logs generated by the Python builtin 'logging' library.
+
+    Taken from
+    https://bitbucket.org/birkenfeld/pygments-main/pull-requests/451/
+    add-python-logging-lexer
```
Comment (CAM-Gerlach) [inline spyder/utils/syntaxhighlighters.py]: The link should not be split between lines, even if it goes over 79 characters.

Probably is short/trivial enough that it doesn't need a mention in NOTICE.txt, but its up to you,

Comment (ccordoba12) [reply spyder/utils/syntaxhighlighters.py]: It's code that was never merged to Pygments, so I don't think it deserves an entry in Notice.txt

Comment (CAM-Gerlach) [reply spyder/utils/syntaxhighlighters.py]: Legally that doesn't really change anything, but practically it might.

**Comment by dalthviz** [comment]
@ccordoba12 a preview:

imagen

imagen

**Comment by ccordoba12** [comment]
Thanks @dalthviz for helping me to finish this one! Things look good but I think we can improve them a little bit. This is how it looks in VSCode:

imagen

So please try to extend the highlighter to match something similar, i.e. to highlight strings, numbers and the field after DEBUG or INFO with different colors. Thanks!

**Comment by ccordoba12** [comment]
And the color for regular text should be black/white, not blue/purple.

**Comment by dalthviz** [comment]
@ccordoba12 a new preview:
imagen
imagen

**Comment by ccordoba12** [comment]
Thanks a lot @dalthviz! Things now look great!

**Pull request merged**